### PR TITLE
ref: Move `event-marker` code into Electron

### LINKED ISSUE #222: Remove Visual Studio and XCode Pre-requisites
I believe these decencies will no longer be needed once we're using electron and electron forge in their latest versions? They should have that functionality built in?

Relies on #220 and #221 first

### PR DESCRIPTION
**Pull request by RobertGemmaJr**
- Moves `event-marker` code into repository directly
- **Removes Visual Studio dependency!**
     - `node-gyp` now rebuilds the correct binaries on Windows even without Visual Studio
 - Windows now installs Python 3.12 to match the Mac install

### CODE REVIEW

File: public/electron/serialPort.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+const SerialPort = require("serialport");
+
+// TODO @RobertGemmaJr: THIS SHOULD ALL BE IN ITS OWN FILE
```
Comment (RobertGemmaJr) [inline public/electron/serialPort.js]:
```suggestion
```

File: public/electron/main.js [hunk 2]
```diff
@@ -8,8 +8,16 @@ const { app, BrowserWindow, ipcMain, dialog } = require("electron");
...
+const { getPort, sendToPort } = require("./serialPort");
+
+// TODO @RobertGemmaJr: Add serialport's MockBinding for the "Continue Anyway": https://serialport.io/docs/guide-testing
+// TODO @RobertGemmaJr: Do more testing with the environment variables - are home/clinic being built correctly?
+
+// TODO @RobertGemmaJr: Serial port doesn't work with node 3.12?
+// TODO @RobertGemmaJr: Only windows installs a Python version?
+// Mac: Homebrew is installing python@3.12 when it installs openjdk
+// pyenv: python@3.10 and @3.12 seem to work okay on windows with pyenv?
+// TODO @RobertGemmaJr: I should handle the change to 'electron-forge dev" at the same time to make sure the rebuild stuff still works correctly
```
Comment (RobertGemmaJr) [inline public/electron/main.js]:
```suggestion
```

File: winget.ps1
Comment (broarr) [inline winget.ps1:1]: you just needed to update python to get rid of visual studio?

**Review by broarr — APPROVED** [review]
This is great! I love a net negative PR!

File: winget.ps1
Comment (RobertGemmaJr) [reply winget.ps1:1]: I needed to update SerialPort, which is why I had to bring all of the event-marker code into the repo

**Pull request merged**